cmd/ftl/integration_test.go

```diff
@@ -143,3 +143,30 @@ func testImportExport(t *testing.T, object string) {
 		}),
 	)
 }
+
+func TestLocalSchemaDiff(t *testing.T) {
+	newVerb := `
+//ftl:verb
+func NewFunction(ctx context.Context, req TimeRequest) (TimeResponse, error) {
+	return TimeResponse{Time: time.Now()}, nil
+}
+`
+	Run(t, "",
+		CopyModule("time"),
+		Deploy("time"),
+		ExecWithOutput("ftl", []string{"schema", "diff"}, func(output string) {
+			assert.Equal(t, "", output)
+		}),
+		EditFile("time", func(bytes []byte) []byte {
+			s := string(bytes)
+			s += newVerb
+			return []byte(s)
+		}, "time.go"),
+		Build("time"),
```

Comment:
Should we implicitly do a build when running a local diff? It does not really seem right, but not doing it also seems like it will result in a high chance of user error.

Reply:
I was initially thinking not, mainly because it's expensive and `ftl dev` might already be running, but I think you're right. It's better to favour correctness, and we can solve the expensiveness problem with future expansions to the build system (ie. proper dependency checks).

Reply:
So if I just call build directly in code the output gets polluted with the build output, which is not really want you want with a diff command. I could try and suppress it (assuming you can do that in go?) but that is also not ideal, or I could try and capture it and only print it on error. In terms of capturing I guess I would need to do something like: https://stackoverflow.com/questions/10473800/in-go-how-do-i-capture-stdout-of-a-function-into-a-string

Or maybe it is not worth building as part of the diff. If we had the ability to detect if a schema is out of date that would probably be enough as we could print a warning to stderr.

Reply:
Makes sense, let's ship it as-is.